beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -370,17 +399,18 @@ func (f *blocksFetcher) requestBlocks(
 	pid peer.ID,
 ) ([]*eth.SignedBeaconBlock, error) {
 	f.Lock()
```

Comment:
This is a global lock that can be blocked by a particular peer's rate limit. In other words, if we are rate limited by one peer, we are blocking all outgoing requests until that peer's rate limit is satisfied. 

Do you want to rearrange this global lock to avoid this type of behavior?

Reply:
Yes, it would be better if we re-arrange the global lock indeed. However, I'd prefer to do it not within the scope of the current PR. Will create another issue to track that improvement.

Reply:
Thanks!